### PR TITLE
Closes #46 - Move should_DeleteOldWorkbasketCleanupJobs_When_InitializingSchedule to AbstractKadaiJobAccTest.java

### LINKED ISSUE #46: Refactore Tests for AbstractTaskanaJob#initializingSchedule
Move the should_Delete....CleanupJobs_When_InitializingSchedule() to AbstractKadaiJobAccTest from these following 3 test classes:

-  HistoryCleanupJobAccTest

- TaskCleanupJobAccTest

- WorkbasketCleanupJobAccTest

The logic for deleting these jobs was moved to AbstractKadaiJob, this is why we should move the test mentioned above

### PR DESCRIPTION
**Pull request by jannikFuellgrafEnvite**
Closes #46 - Move should_DeleteOldWorkbasketCleanupJobs_When_InitializingSchedule to AbstractKadaiJobAccTest.java

### Thanks for your PR! Please fill out the following list :)

---

- [ ] I put the ticket or multiple tickets in review
- [x] Commit message format → Closes #&lt;Issue Number&gt; - Your commit message.
- [x] Sonarcloud link : https://sonarcloud.io/summary/new_code?id=jannikFuellgrafEnvite_kadai&branch=kadai%2F46
- [x] No documentation update needed
- [ ] Link to PR with documentation update: \<add the link here>
- [x] No Release Notes needed
- [ ] Release Notes :

```

```

### CODE REVIEW

File: lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java [hunk 1]
```diff
@@ -106,37 +100,6 @@ void shouldNotCleanWorkbasketMarkedForDeletionIfWorkbasketHasTasks() throws Exce
...
-      kadaiEngine.getJobService().createJob(job);
-      job.setType(TaskRefreshJob.class.getName());
-      kadaiEngine.getJobService().createJob(job);
-      job.setType(ClassificationChangedJob.class.getName());
-      kadaiEngine.getJobService().createJob(job);
-    }
-
-    List<ScheduledJob> jobsToRun = getJobMapper(kadaiEngine).findJobsToRun(Instant.now());
-
-    assertThat(jobsToRun).hasSize(30);
```
Comment (CRoberto1926) [inline lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java:125]: Why did you remove this assertion?

**Review by CRoberto1926 — COMMENTED** [review]
What about this comment?

// Dirty Hack, please refactor me with ticket https://github.com/kadai-io/kadai/issues/46

File: lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java (hunk 1 above)
Comment (jannikFuellgrafEnvite) [reply lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java:125]: I deleted the test function because I included the test case in the AbstractKadaiJobAccTest class, as required by the issue. James gave me a similar issue, which I followed: https://github.com/Taskana/taskana/pull/2470/files#diff-a9864bf32971d758536dbe9eb5f708dd09072bcc33be23bbfef2f5e7d982af22

**Comment by jannikFuellgrafEnvite** [comment]
> What about this comment?
> 
> // Dirty Hack, please refactor me with ticket #46

James told me to ignore it and just refactor the test function

File: lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java (hunk 1 above)
Comment (CRoberto1926) [reply lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java:125]: ok so it is fine to not test size 30 anymore?

Comment (jamesrdi) [reply lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java:125]: This test is not relevant anymore, the test has been moved to the AbstractKadaiJobAccTest in the should_DeleteOldCleanupJobs_When_InitializingSchedule() test

Comment (CRoberto1926) [reply lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java:125]: Yes that's clear. What I mean is: should the test `should_DeleteOldCleanupJobs_When_InitializingSchedule` in `AbstractKadaiJobAccTest` also `assertThat(jobsToRun).hasSize(30)` or not? And if not, why?

Comment (jamesrdi) [reply lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java:125]: I see your point now. Technically we don't need to check again if `jobsToRun` has size 30, since we know that after the for loop, 30 jobs will be initiated, 10 from TaskRefreshJob, 10 from ClassificationChangedJob, and 10 from the CleanUpJob, so asserting the `jobsToRun` is redundant

Comment (CRoberto1926) [reply lib/kadai-core/src/test/java/acceptance/jobs/WorkbasketCleanupJobAccTest.java:125]: Perfect, thanks for the explanation 👍

**Review by CRoberto1926 — APPROVED** [review]
